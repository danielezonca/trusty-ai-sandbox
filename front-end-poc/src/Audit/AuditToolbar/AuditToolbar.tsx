import React, { useRef } from "react";
import {
  DataToolbar,
  DataToolbarContent,
  DataToolbarItem,
  DataToolbarItemVariant,
} from "@patternfly/react-core/dist/js/experimental";
import { Button, ButtonVariant, InputGroup, TextInput } from "@patternfly/react-core";
import { SearchIcon } from "@patternfly/react-icons";
import FromFilter from "../FromFilter/FromFilter";
import ToFilter from "../ToFilter/ToFilter";
import PaginationContainer from "../PaginationContainer/PaginationContainer";

type AuditToolbarProps = {
  setSearchString: (searchString: string) => void;
  fromDate: string;
  setFromDate: (fromDate: string) => void;
  toDate: string;
  setToDate: (toDate: string) => void;
  total: number;
  pageSize: number;
  page: number;
  setPage: (page: number) => void;
  setPageSize: (pageSize: number) => void;
};

const AuditToolbar = (props: AuditToolbarProps) => {
  const {
    setSearchString,
    fromDate,
    setFromDate,
    toDate,
    setToDate,
    total,
    pageSize,
    page,
    setPage,
    setPageSize,
  } = props;

  const searchField = useRef<HTMLInputElement>(null);
  const onSearchSubmit = (): void => {
    if (searchField && searchField.current) setSearchString(searchField.current.value);
  };
  const onSearchEnter = (event: React.KeyboardEvent): void => {
    if (searchField && searchField.current && event.key === "Enter") {
      setSearchString(searchField.current.value);
    }
  };
  return (
    <DataToolbar id="audit-list-top-toolbar" style={{ marginBottom: "var(--pf-global--spacer--lg)" }}>
      <DataToolbarContent>
        <DataToolbarItem variant="label">Search</DataToolbarItem>
        <DataToolbarItem>
          <InputGroup>
            <TextInput
              name="audit-search-input"
              ref={searchField}
              id="audit-search-input"
              type="search"
              aria-label="search executions"
              onKeyDown={onSearchEnter}
            />
            <Button
              id="audit-search"
              variant={ButtonVariant.control}
              aria-label="search button for search input"
              onClick={onSearchSubmit}>
              <SearchIcon />
            </Button>
          </InputGroup>
        </DataToolbarItem>
        <DataToolbarItem variant="label">From</DataToolbarItem>
        <DataToolbarItem>
          <FromFilter fromDate={fromDate} onFromDateUpdate={setFromDate} maxDate={toDate} />
        </DataToolbarItem>
        <DataToolbarItem variant="label">To</DataToolbarItem>
        <DataToolbarItem>
          <ToFilter toDate={toDate} onToDateUpdate={setToDate} minDate={fromDate} />
        </DataToolbarItem>
        <DataToolbarItem variant={DataToolbarItemVariant.pagination}>
          <PaginationContainer
            total={total}
            page={page}
            pageSize={pageSize}
            onSetPage={setPage}
            onSetPageSize={setPageSize}
            paginationId="audit-list-top-pagination"
          />
        </DataToolbarItem>
      </DataToolbarContent>
    </DataToolbar>
  );
};

export default AuditToolbar;
