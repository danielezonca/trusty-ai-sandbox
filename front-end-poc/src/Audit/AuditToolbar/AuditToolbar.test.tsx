import React from "react";
import AuditToolbar from "./AuditToolbar";
import { mount, shallow } from "enzyme";

describe("Audit toolbar", () => {
  test("renders", () => {
    const wrapper = renderAuditToolbar("shallow");
    expect(wrapper).toMatchSnapshot();
  });

  test("allows free text search", () => {
    const setSearchString = jest.fn();
    const searchString = "#12345";
    const wrapper = renderAuditToolbar("mount", { setSearchString });
    const searchInput = wrapper.find("input#audit-search-input");
    const searchButton = wrapper.find("button#audit-search");
    // searchInput.at(0).instance().value = "";
    const inputNode = searchInput.getDOMNode<HTMLInputElement>();

    inputNode.value = searchString;
    searchButton.simulate("click");

    expect(setSearchString).toBeCalledTimes(1);
    expect(setSearchString).toBeCalledWith(searchString);
  });
});

const renderAuditToolbar = (method: "shallow" | "mount", props?: object) => {
  const defaultProps = {
    setSearchString() {},
    fromDate: "2020-01-01",
    setFromDate() {},
    toDate: "2020-02-01",
    setToDate() {},
    total: 20,
    page: 1,
    pageSize: 10,
    setPage() {},
    setPageSize() {},
    ...props,
  };
  if (method === "shallow") {
    return shallow(<AuditToolbar {...defaultProps} />);
  } else {
    return mount(<AuditToolbar {...defaultProps} />);
  }
};
