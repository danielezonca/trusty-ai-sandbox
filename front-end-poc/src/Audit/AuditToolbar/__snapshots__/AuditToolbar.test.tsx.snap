// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audit toolbar renders 1`] = `
<DataToolbar
  id="audit-list-top-toolbar"
  style={
    Object {
      "marginBottom": "var(--pf-global--spacer--lg)",
    }
  }
>
  <DataToolbarContent
    breakpointMods={Array []}
    isExpanded={false}
    showClearFiltersButton={false}
  >
    <DataToolbarItem
      variant="label"
    >
      Search
    </DataToolbarItem>
    <DataToolbarItem>
      <InputGroup>
        <ForwardRef
          aria-label="search executions"
          id="audit-search-input"
          name="audit-search-input"
          onKeyDown={[Function]}
          type="search"
        />
        <Component
          aria-label="search button for search input"
          id="audit-search"
          onClick={[Function]}
          variant="control"
        >
          <SearchIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
            title={null}
          />
        </Component>
      </InputGroup>
    </DataToolbarItem>
    <DataToolbarItem
      variant="label"
    >
      From
    </DataToolbarItem>
    <DataToolbarItem>
      <FromFilter
        fromDate="2020-01-01"
        maxDate="2020-02-01"
        onFromDateUpdate={[Function]}
      />
    </DataToolbarItem>
    <DataToolbarItem
      variant="label"
    >
      To
    </DataToolbarItem>
    <DataToolbarItem>
      <ToFilter
        minDate="2020-01-01"
        onToDateUpdate={[Function]}
        toDate="2020-02-01"
      />
    </DataToolbarItem>
    <DataToolbarItem
      variant="pagination"
    >
      <PaginationContainer
        onSetPage={[Function]}
        onSetPageSize={[Function]}
        page={1}
        pageSize={10}
        paginationId="audit-list-top-pagination"
        total={20}
      />
    </DataToolbarItem>
  </DataToolbarContent>
</DataToolbar>
`;
